packages/material-ui-lab/src/TreeItem/TreeItem.js

```diff
@@ -1,102 +1,138 @@
+import {
+  alpha,
+  experimentalStyled,
+  unstable_useThemeProps as useThemeProps,
+} from '@material-ui/core/styles';
 import { ownerDocument, useForkRef, unsupportedProp } from '@material-ui/core/utils';
+import { unstable_composeClasses as composeClasses } from '@material-ui/unstyled';
 import TreeViewContext from '../TreeView/TreeViewContext';
 import { DescendantProvider, useDescendant } from '../TreeView/descendants';
 import TreeItemContent from './TreeItemContent';
+import treeItemClasses, { getTreeItemUtilityClass } from './treeItemClasses';
 
-export const styles = (theme) => ({
-  /* Styles applied to the root element. */
-  root: {
-    listStyle: 'none',
-    margin: 0,
-    padding: 0,
-    outline: 0,
+const useUtilityClasses = (styleProps) => {
+  const { classes } = styleProps;
+
+  const slots = {
+    root: ['root'],
+    content: ['content'],
+    expanded: ['expanded'],
+    selected: ['selected'],
+    focused: ['focused'],
+    disabled: ['disabled'],
+    iconContainer: ['iconContainer'],
+    label: ['label'],
+    group: ['group'],
+  };
+
+  return composeClasses(slots, getTreeItemUtilityClass, classes);
+};
+
+const TreeItemRoot = experimentalStyled(
+  'li',
+  {},
+  {
+    name: 'MuiTreeItem',
+    slot: 'Root',
+    overridesResolver: (props, styles) => styles.root,
+  },
+)({
+  listStyle: 'none',
+  margin: 0,
+  padding: 0,
+  outline: 0,
+});
+
+const StyledTreeItemContent = experimentalStyled(
+  TreeItemContent,
+  {},
+  {
+    name: 'MuiTreeItem',
+    slot: 'Content',
+    overridesResolver: (props, styles) => {
+      const { styleProps } = props;
+      return deepmerge(
+        {
+          ...(styles.iconContainer && {
+            [`.${treeItemClasses.iconContainer}`]: styles.iconContainer,
+          }),
+          ...(styles.label && {
+            [`.${treeItemClasses.label}`]: styles.label,
+          }),
+          ...(styleProps.expanded && styles.expanded),
+          ...(styleProps.selected && styles.selected),
+          ...(styleProps.focused && styles.focused),
+          ...(styleProps.disabled && styles.disabled),
+        },
+        styles.content || {},
+      );
+    },
   },
-  /* Styles applied to the transition component. */
-  group: {
-    margin: 0,
-    padding: 0,
-    marginLeft: 17,
+)(({ theme, styleProps }) => ({
+  padding: '0 8px',
+  width: '100%',
+  display: 'flex',
+  alignItems: 'center',
+  cursor: 'pointer',
+  WebkitTapHighlightColor: 'transparent',
+  '&:hover': {
+    backgroundColor: theme.palette.action.hover,
+    // Reset on touch devices, it doesn't add specificity
+    '@media (hover: none)': {
+      backgroundColor: 'transparent',
+    },
   },
-  /* Styles applied to the content element. */
-  content: {
-    padding: '0 8px',
-    width: '100%',
-    display: 'flex',
-    alignItems: 'center',
-    cursor: 'pointer',
-    WebkitTapHighlightColor: 'transparent',
+  ...(styleProps.disabled && {
```

Comment:
For `disabled` (and other pseudo states) we always use the pseudo-classes and we are bumping the specificity to be aligned with the CSS specification, more can be fined https://next.material-ui.com/customization/how-to-customize/#pseudo-classes

You can follow any other component, for example https://github.com/mui-org/material-ui/blob/next/packages/material-ui/src/Button/Button.js#L162